### PR TITLE
Fixes #559 : Refactor API endpoints into individual files

### LINKED ISSUE #559: Refactor API endpoints into individual files
All the handlers for every API endpoint is located in one file - https://github.com/tattle-made/Uli/blob/main/browser-extension/api-server/index.js
This was good when we started out, but its difficult to maintain as the app grows and new maintainers join. Lets refactor them into endpoint specific files. 

Please suggest your idea for how you would refactor the endpoints and we can critique it. When approved you can start the implementation and make a PR

### PR DESCRIPTION
**Pull request by hardik-pratap-singh**
---
name: Fix for Issue #559
about: Refactored APIs into MVC architecture
labels: priority:medium, enhancement, backend

---

**Describe the PR**

Earlier, all the APIs were present in single file called `index.js` which was really difficult to maintain for the maintainers. Now this PR has divided those endpoints into files and these files contains similar types of endpoints. Also, this PR has separated the names of endpoints along with their implementations in separate folders called `routes` and `controllers` respectively.

**Steps to test the PR**

Modified File : `browser-extension/api-server/index.js`
New Directory : `browser-extension/api-server/routes`
New Directory : `browser-extension/api-server/controllers`

**Expected behavior**

`browser-extension/api-server/routes`
image

`browser-extension/api-server/controllers`
image

Now, `browser-extension/api-server/index.js` looks something like this : 
image

### CODE REVIEW

File: browser-extension/plugin/src/ui-components/pages/Slur.jsx
Comment (dennyabrain) [inline browser-extension/plugin/src/ui-components/pages/Slur.jsx:1]: Is it possible to remove this file? I think this is a remnant of the other UI PR?

File: browser-extension/plugin/manifest.json
Comment (dennyabrain) [inline browser-extension/plugin/manifest.json:1]: Make sure that the "localhost" urls are not in there and the actual uli urls are there.
Even better, since this change is not related to the PR you have made, it would be nice to remove this from this PR altogether.

**Comment by dennyabrain** [comment]
@hardik-pratap-singh This looks great! Since its a big change and we don't have automated tests (something we need to fix) it will take us a while to review this. Give us time till next week to get back to you on this.

**Comment by aatmanvaidya** [comment]
I have reviewed this PR, things look good to me

Its read to merge

These are the following things I tested on `Chrome`, `Brave` and `Firefox`
1. Slur Replacement
2. Slur Crowdsource feature (CRUD)
3. Right Click and add slur (CRUD)
4. Preferences Tab
  -  Manually add a word to the slur list and upon saving, slur replacement should work for that word
  - The enable/disable slur replacement checkbox
  - Adding email 
  - changing language
5. Reset Account 
7. Data in the Resources tab is also shown correctly!

@dennyabrain things look good to me, can you take a final look and merge the PR

**Review by aatmanvaidya — APPROVED** [review]
